pkg/reconciler/revision/resources/deploy_test.go

```diff
@@ -776,19 +778,67 @@ func TestMakeDeployment(t *testing.T) {
 		want: appsv1deployment(func(deploy *appsv1.Deployment) {
 			deploy.Spec.ProgressDeadlineSeconds = ptr.Int32(42)
 		}),
+	}, {
+		name:                "cluster initial scale",
+		clusterInitialScale: ptr.Int32(int32(10)),
+		rev: revision("bar", "foo",
+			withoutLabels,
+			func(revision *v1.Revision) {
+				container(revision.Spec.GetContainer(),
+					withReadinessProbe(corev1.Handler{
+						TCPSocket: &corev1.TCPSocketAction{
+							Host: "127.0.0.1",
+							Port: intstr.FromInt(12345),
+						},
+					}),
+				)
+			},
+		),
+		want: appsv1deployment(func(deploy *appsv1.Deployment) {
+			deploy.Spec.Replicas = ptr.Int32(int32(10))
+		}),
+	}, {
+		name:                "cluster initial scale override by revision initial scale",
+		clusterInitialScale: ptr.Int32(int32(10)),
+		rev: revision("bar", "foo",
+			withoutLabels,
+			func(revision *v1.Revision) {
+				container(revision.Spec.GetContainer(),
+					withReadinessProbe(corev1.Handler{
+						TCPSocket: &corev1.TCPSocketAction{
+							Host: "127.0.0.1",
+							Port: intstr.FromInt(12345),
+						},
+					}),
+				)
+				revision.ObjectMeta.Annotations = map[string]string{autoscaling.InitialScaleAnnotationKey: "20"}
+			},
+		),
+		want: appsv1deployment(func(deploy *appsv1.Deployment) {
+			deploy.Spec.Replicas = ptr.Int32(int32(20))
+			deploy.Spec.Template.ObjectMeta.Annotations = map[string]string{autoscaling.InitialScaleAnnotationKey: "20"}
+			deploy.ObjectMeta.Annotations = map[string]string{autoscaling.InitialScaleAnnotationKey: "20"}
+		}),
 	}}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			// Tested above so that we can rely on it here for brevity.
+			asConfigPtr, err := autoscalerconfig.NewConfigFromMap(map[string]string{})
```

Comment:
wondering if we could use the same pattern here as the other configs, i.e. similar to how we test overriding `ProgressDeadline` already in the table. Just use the zero value of autoscalerconfig.Config as the base case (i.e. add a `ac autoscalerconfig.Config` field to the table) and specify an override it explicitly in the table just for these two new rows.